src/data/repositories/DataSetD2Api.ts

```diff
@@ -1,91 +1,102 @@
 import { D2Api, MetadataPick } from "$/types/d2-api";
-import { FutureData, apiToFuture } from "$/data/api-futures";
-import { AccessData, AccessType, CoreCompetency, DataSet } from "$/domain/entities/DataSet";
+import { apiToFuture } from "$/data/api-futures";
+import {
+    AccessData,
+    AccessType,
+    CoreCompetency,
+    DataSet,
+    OrgUnit,
+    Permissions,
+} from "$/domain/entities/DataSet";
 import { Paginated } from "$/domain/entities/Paginated";
 import { GetDataSetOptions } from "$/domain/repositories/DataSetRepository";
-import { Future } from "$/domain/entities/generic/Future";
+import { Future, FutureData } from "$/domain/entities/generic/Future";
 import { Maybe } from "$/utils/ts-utils";
-import { Id, OctalNotationPermission, Ref } from "$/domain/entities/Ref";
+import { Id } from "$/domain/entities/Ref";
 import { Permission } from "$/domain/entities/Permission";
 import _ from "$/domain/entities/generic/Collection";
 import { Project } from "$/domain/entities/Project";
 import { D2ApiCategoryOption } from "$/data/repositories/D2ApiCategoryOption";
-import { MetadataItem } from "$/domain/entities/MetadataItem";
+import { D2ApiConfig, D2Config } from "$/data/repositories/D2ApiConfig";
 
 export class DataSetD2Api {
     private d2ApiCategoryOption: D2ApiCategoryOption;
-    constructor(private api: D2Api, private metadata: MetadataItem) {
+    private d2ApiConfig: D2ApiConfig;
+
+    constructor(private api: D2Api) {
         this.d2ApiCategoryOption = new D2ApiCategoryOption(this.api);
+        this.d2ApiConfig = new D2ApiConfig(this.api);
     }
 
     get(options: D2DataSetOptions): FutureData<Paginated<DataSet>> {
```

Comment:
In general, we try to keep methods short and descriptive, and that's special relevant for public methods. In a glance, we should be able to tell what steps the public method is performing. We can abstract operations to instance methods.